### PR TITLE
Improve docs layout

### PR DESCRIPTION
**Pull request by adriencaccia**
Changes navbar layout:
image

Change docs sidebar layout and add index cards:
image

### CODE REVIEW

**Comment by adriencaccia** [comment]
I think the docs look better this way, @fargito WDYT ?

File: user-docs/documentation/docs/explanations/1-swarmion-philosophy/index.md [hunk 1]
```diff
@@ -1,9 +1,10 @@
----
-sidebar_position: 0
----
+import DocCardList from '@theme/DocCardList';
+import {useCurrentSidebarCategory} from '@docusaurus/theme-common';
 
 # Swarmion philosophy
 
+<DocCardList items={useCurrentSidebarCategory().items}/>
```
Comment (adriencaccia) [inline user-docs/documentation/docs/explanations/1-swarmion-philosophy/index.md:6]: Rename into `Why swarmion ?`
Modify the introduction home page to link to it

File: user-docs/documentation/references/index.md [hunk 2]
```diff
@@ -1,5 +1,5 @@
 ---
-sidebar_position: 1
+sidebar_position: 0
 ---
 
 # References
```
Comment (adriencaccia) [inline user-docs/documentation/references/index.md:5]: Add a coming soon comment

**Comment by fargito** [comment]
This is great, thanks a lot!

**Review by fargito — APPROVED** [review]

**Pull request merged**